l1-contracts/test/TestERC20.t.sol

```diff
@@ -1,24 +1,90 @@
-pragma solidity ^0.8.18;
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.27;
 
-import "forge-std/Test.sol";
+import {Test} from "forge-std/Test.sol";
 import {TestERC20} from "@aztec/mock/TestERC20.sol";
 
+// solhint-disable comprehensive-interface
+// solhint-disable func-name-mixedcase
+
 contract TestERC20Test is Test {
-  TestERC20 testERC20;
+  TestERC20 internal testERC20;
 
   function setUp() public {
     testERC20 = new TestERC20("test", "TEST", address(this));
   }
 
+  function test_addRemoveMinter(address _minter) public {
+    vm.assume(_minter != address(this));
+    vm.assume(_minter != address(0));
+
+    vm.prank(address(_minter));
+    vm.expectRevert();
+    testERC20.mint(address(this), 100);
+
+    vm.expectEmit();
+    emit TestERC20.MinterAdded(_minter);
+    testERC20.addMinter(address(_minter));
+
+    vm.prank(address(_minter));
+    testERC20.mint(address(this), 100);
+
+    vm.expectEmit();
```

Comment:
You need to update the `expectEmit` to ensure that it is actually checking that the values in the event makes sense and that it is from the correct address.

As I recall, the first 3 bools are for the indexed values, then for the data and then the address that should emit.

```solidity
vm.expectEmit(true, true, true, true, address(testERC20));
```